dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsImplCommon.java

```diff
@@ -148,23 +148,25 @@ private static String extractHTTPPrefix(String gitUrl, String reference, String
 
         String url;
         String newID = getNewId(container);
-        if (newID == null) {
+        url = getUrlFromId(config, newID);
+        if (url == null) {
             return null;
-        } else {
-            try {
-                String baseURL = baseURL(config);
-                url = getUrl(newID, baseURL);
-            } catch (URISyntaxException | UnsupportedEncodingException e) {
-                LOG.error("Could not construct URL for our container with id: " + container.getId());
-                return null;
-            }
         }
         // TODO: hook this up to a type field in our DB?
         io.swagger.model.Tool tool = new io.swagger.model.Tool();
         tool = setGeneralToolInfo(tool, container);
         tool.setId(newID);
         tool.setUrl(url);
-
+        String checkerWorkflowPath = getCheckerWorkflowPath(config, container);
+        if (checkerWorkflowPath == null) {
+            checkerWorkflowPath = "";
```

Comment:
Is there a reason we don't do null?

Reply:
Swagger specification doesn't like it.  In fact, people defined x-nullable into the Swagger specifications to indicate that the properties are nullable

Reply:
Yeah, I think the cwl people ran into this with their python validator too